payjoin/src/receive/v1/mod.rs

```diff
@@ -553,6 +553,64 @@ impl WantsInputs {
             change_vout: self.change_vout,
         }
     }
+
+    /// Select inputs optimizing for consolidation when fees are low
+    pub(crate) fn try_consolidate(
+        &self,
+        candidate_inputs: impl IntoIterator<Item = InputPair>,
+        dust_limit: Option<Amount>,
+        max_inputs: Option<usize>,
+        target_value: Option<Amount>,
+        fee_rate_threshold: Option<FeeRate>,
+    ) -> Result<Vec<InputPair>, SelectionError> {
+        // Check current fee rate
+        let current_fee = self.payjoin_psbt.fee().expect("fee exists");
+        let current_fee_rate = FeeRate::from_sat_per_vb(
+            current_fee.to_sat() / self.payjoin_psbt.unsigned_tx.weight().to_vbytes_ceil(),
+        );
+
+        // If fee rate is above threshold, return error
+        if let (Some(current), Some(threshold)) = (current_fee_rate, fee_rate_threshold) {
+            if current > threshold {
+                return Err(InternalSelectionError::FeeRateTooHighForConsolidation.into());
+            }
+        }
+
+        // Filter and collect valid inputs above dust limit
+        let dust = dust_limit.unwrap_or(DEFAULT_DUST_LIMIT);
+        let mut valid_inputs: Vec<InputPair> = candidate_inputs
+            .into_iter()
+            .filter(|input| input.previous_txout().value >= dust)
+            .collect();
+
+        if valid_inputs.is_empty() {
+            return Err(InternalSelectionError::Empty.into());
+        }
+
+        // Sort inputs by value ascending
+        valid_inputs.sort_by_key(|input| input.previous_txout().value);
```

Comment:
If you're going to leave a comment about sorting, tell me *why* you're doing it. (presumably to make the for loop below exit early)